lang/cs/Org.Apache.REEF.IMRU.Tests/JobLifecycleManagerTest.cs

```diff
@@ -105,6 +111,35 @@ public void JobLifeCyclemangerNoObserversDoesNotCheckForSignal()
             AssertCancelEvent(observer, false);
         }
 
+        [Fact]
+        [Trait("Description", "Verify that manager stops checking for cancellation signal after all observers unsubscribed.")]
+        public void JobLifeCyclemangerChecksStopsCancellationChecksWhenAllObserversUnsubscribed()
```

Comment:
yes, that was confusing name. I rephrased, but kept somewhat long. :-)
I think for test naming is more important even if it's long. (It would be easier to read if we allow underscores in names.)